common/namespace/registry.go

```diff
@@ -27,95 +27,10 @@
 package namespace
 
 import (
-	"context"
-	"sync"
-	"sync/atomic"
-	"time"
-
-	"go.temporal.io/api/serviceerror"
-	expmaps "golang.org/x/exp/maps"
-
-	"go.temporal.io/server/common/cache"
-	"go.temporal.io/server/common/clock"
-	"go.temporal.io/server/common/dynamicconfig"
-	"go.temporal.io/server/common/headers"
-	"go.temporal.io/server/common/log"
-	"go.temporal.io/server/common/log/tag"
-	"go.temporal.io/server/common/metrics"
-	"go.temporal.io/server/common/persistence"
 	"go.temporal.io/server/common/pingable"
-	"go.temporal.io/server/internal/goro"
-)
-
-// ReplicationPolicy is the namespace's replication policy,
-// derived from namespace's replication config
-type ReplicationPolicy int
-
-const (
-	// ReplicationPolicyOneCluster indicate that workflows does not need to be replicated
-	// applicable to local namespace & global namespace with one cluster
-	ReplicationPolicyOneCluster ReplicationPolicy = 0
-	// ReplicationPolicyMultiCluster indicate that workflows need to be replicated
-	ReplicationPolicyMultiCluster ReplicationPolicy = 1
-)
-
-const (
-	cacheMaxSize = 64 * 1024
-	cacheTTL     = 0 // 0 means infinity
-	// CacheRefreshFailureRetryInterval is the wait time
-	// if refreshment encounters error
-	CacheRefreshFailureRetryInterval = 1 * time.Second
-	CacheRefreshPageSize             = 1000
-	readthroughCacheTTL              = 1 * time.Second // represents minimum time to wait before trying to readthrough again
-	readthroughTimeout               = 3 * time.Second
-)
-
-const (
-	stopped int32 = iota
-	starting
-	running
-	stopping
-)
-
-var (
-	cacheOpts = cache.Options{
-		TTL: cacheTTL,
-	}
-	readthroughNotFoundCacheOpts = cache.Options{
-		TTL: readthroughCacheTTL,
-	}
 )
 
 type (
-	// Clock provides timestamping to Registry objects
-	Clock interface {
-		// Now returns the current time.
-		Now() time.Time
-	}
-
-	// Persistence describes the durable storage requirements for a Registry
-	// instance.
-	Persistence interface {
-
-		// GetNamespace reads the state for a single namespace by name or ID
-		// from persistent storage, returning an instance of
-		// serviceerror.NamespaceNotFound if there is no matching Namespace.
-		GetNamespace(
-			context.Context,
-			*persistence.GetNamespaceRequest,
-		) (*persistence.GetNamespaceResponse, error)
-
-		// ListNamespaces fetches a paged set of namespace persistent state
-		// instances.
-		ListNamespaces(
-			context.Context,
-			*persistence.ListNamespacesRequest,
-		) (*persistence.ListNamespacesResponse, error)
-
-		// GetMetadata fetches the notification version for Temporal namespaces.
-		GetMetadata(context.Context) (*persistence.GetMetadataResponse, error)
-	}
-
 	// PrepareCallbackFn is function to be called before CallbackFn is called,
 	// it is guaranteed that PrepareCallbackFn and CallbackFn pair will be both called or non will be called
 	PrepareCallbackFn func()
```

Comment:
this type and several others are duplicated in the new file

Reply:
Hahah! Thats funny. Its not used anywhere. I was depending on import errors to help me choose where to put things and this didn't come up because we use it twice.